### ISSUE TITLE
The list of updates no longer appears

### ISSUE THREAD

**Issue by rbatllet** [Homebrew/brew#13412]
### `brew config` output

```shell
HOMEBREW_VERSION: 3.5.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2258ba5797c1ea8149a49673b5c080011c366237
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f32c115b61f74b1327eddbeb163acfa8321ee78b
Core tap last commit: 5 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 13.0.0 build 1300
Git: 2.36.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6.6-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
```

### `brew doctor` output

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The filesystem on /System/Volumes/Data appears to be case-sensitive.
The default macOS filesystem is case-insensitive. Please report any apparent problems.
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Since the Ruby update by Brew a few days ago, the list of updates no longer appears when I run "brew update".

### What happened (include all command output)?

% brew update && brew upgrade && brew cleanup

Updated 2 taps (homebrew/core and homebrew/cask).

### What did you expect to happen?

I hope that the list of updates will be displayed as it has always been.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
% brew update && brew upgrade && brew cleanup

Updated 2 taps (homebrew/core and homebrew/cask).
```

**Comment by apainintheneck** [comment]
There was a change made in 3.5.0 to the update command. It now defaults to only showing output for installed packages. The environment variable `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE` can be set if you would like to return to the old behavior. Note that there is currently a bug with the old behavior where it doesn't print new formulae and casks (modified formulae and casks show up fine though) but that should be resolved soon (#13403).

More information can be found in the [3.5.0 release notes](https://brew.sh/2022/06/06/homebrew-3.5.0/) and in this [recent discussion thread](https://github.com/orgs/Homebrew/discussions/3353).

**Comment by MikeMcQuaid** [comment]
> I think it is a bad idea to hide the list of updates as it becomes difficult to find new formulas this way.

You can adjust this behaviour for yourself if desired with `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE`. We may consider adjusting this behaviour for e.g. new formulae alone if we get more feedback like yours.
